src/filter.js

```diff
@@ -36,12 +35,11 @@ var keys = require('./keys');
 module.exports = _curry2(_dispatchable('filter', _xfilter, function(pred, filterable) {
   return (
     _isObject(filterable) ?
-      _reduce(function(acc, key) {
+      _accumObj(function(key, value, newObject) {
         if (pred(filterable[key])) {
-          acc[key] = filterable[key];
+          newObject[key] = value;
         }
-        return acc;
-      }, {}, keys(filterable)) :
+      }, filterable) :
     // else
```

Comment:
The same applies to all those functions that already traverse objects this way (e.g. ``evolve``). I guess copying over these properties doesn't break much. Also note that current implementation of ``keys``, in ES5+ environments, only takes care of the Safari enum bug.